src/core/webidl-index.js

```diff
@@ -29,7 +29,23 @@ export function run() {
     }
     idlIndexSec.insertAdjacentElement("afterbegin", header);
   }
-  if (!document.querySelector("pre.idl")) {
+
+  //filter out the IDL marked with class="idlExclude" and the IDL in non-normative sections
+  function idlFilter(idl) {
+    if (
+      idl.classList.contains("idlExclude") ||
+      idl.parentElement.classList.contains("informative")
+    ) {
+      return false;
+    }
+    return true;
+  }
+
+  const idlIndex = Array.from(document.querySelectorAll("pre.def.idl")).filter(
+    idlFilter
+  );
+
+  if (!document.querySelector("pre.idl") || idlIndex.length === 0) {
```

Comment:
I think you should be able to exclude/filter in this CSS query itself by using `:not(.exclude)`, then you might not need the filter above. 

Make sure you add some tests too :)

Reply:
> then you might not need the filter above.

The exclude thing can be done with `:not()`, but can we use pure CSS selector to blacklist a parent?

Reply:
@saschanaz is right, we can’t go to the parent. So at least `:not()` first then filter on `.closest()`.

Reply:
(`:not(.informative) > pre.idl:not(.exclude)` will work if we are only to check direct ascendants, but probably `closest()` is better here.)